### PR TITLE
rio: v0.0.28 -> v0.0.29

### PR DESCRIPTION
**Pull request by TornaxO7**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by otavio — APPROVED** [review]

**Review by otavio — CHANGES_REQUESTED** [review]
Please revert formatting changes.

**Review by otavio — COMMENTED** [review]
Please add your `githubId`.

**Review by otavio — APPROVED** [review]

**Review by otavio — APPROVED** [review]

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -18526,6 +18526,12 @@
     githubId = 1486805;
     name = "Toon Nolten";
   };
+  tornax = {
```
Comment (AndersonTorres) [inline maintainers/maintainer-list.nix:18529]: **Flip**

File: pkgs/applications/terminal-emulators/rio/default.nix [hunk 2]
```diff
@@ -113,7 +113,7 @@ rustPlatform.buildRustPackage rec {
     description = "A hardware-accelerated GPU terminal emulator powered by WebGPU";
     homepage = "https://raphamorim.io/rio";
     license = lib.licenses.mit;
-    maintainers = with lib.maintainers; [ otavio oluceps ];
+    maintainers = with lib.maintainers; [ tornax otavio oluceps ];
```
Comment (AndersonTorres) [inline pkgs/applications/terminal-emulators/rio/default.nix:116]: **Split**

File: pkgs/applications/terminal-emulators/rio/default.nix [hunk 3]
```diff
@@ -28,37 +28,38 @@
...
-  ] ++ lib.optionals withX11 [
-    libX11
-    libXcursor
-    libXi
-    libXrandr
-    libxcb
-  ] ++ lib.optionals withWayland [
-    wayland
-  ];
+  rlinkLibs =
```
Comment (AndersonTorres) [inline pkgs/applications/terminal-emulators/rio/default.nix:31]: Why?

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/applications/terminal-emulators/rio/default.nix (hunk 2 above)
Comment (TornaxO7) [reply pkgs/applications/terminal-emulators/rio/default.nix]: done

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (AndersonTorres) [reply maintainers/maintainer-list.nix:18529]: The commit adding you as new maintainer should be the first on the stack.
Also, it should contain only this.

Comment (TornaxO7) [reply maintainers/maintainer-list.nix:18529]: done

File: pkgs/applications/terminal-emulators/rio/default.nix [hunk 4]
```diff
@@ -112,7 +112,11 @@ rustPlatform.buildRustPackage rec {
     description = "A hardware-accelerated GPU terminal emulator powered by WebGPU";
     homepage = "https://raphamorim.io/rio";
     license = lib.licenses.mit;
-    maintainers = with lib.maintainers; [ otavio oluceps ];
+    maintainers = with lib.maintainers; [
```
Comment (AndersonTorres) [inline pkgs/applications/terminal-emulators/rio/default.nix:116]: *split*

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/applications/terminal-emulators/rio/default.nix (hunk 4 above)
Comment (TornaxO7) [reply pkgs/applications/terminal-emulators/rio/default.nix:116]: I... don't understand what you mean. I though with "split" you meant "split the list into several lines".

**Review by otavio — APPROVED** [review]
LGTM 👍🏻

**Review by AndersonTorres — APPROVED** [review]

**Review by otavio — APPROVED** [review]

**Comment by thiagokokada** [comment]
Build is failing in Darwin, is this expected?

**Comment by otavio** [comment]
@thiagokokada do you have the log? I don't have access to Darwin device.

**Comment by thiagokokada** [comment]
> @thiagokokada do you have the log? I don't have access to Darwin device.

I am looking at the Ofborg logs.

**Comment by TornaxO7** [comment]
hm... I have no idea how to fix this

File: pkgs/applications/terminal-emulators/rio/default.nix [hunk 5]
```diff
@@ -31,6 +31,7 @@ let
   rlinkLibs = if stdenv.isDarwin then [
     darwin.libobjc
     darwin.apple_sdk_11_0.frameworks.AppKit
+    darwin.apple_sdk_11_0.frameworks.VisionKit
```
Comment (thiagokokada) [inline pkgs/applications/terminal-emulators/rio/default.nix:34]: This doesn't seem to exist.

**Review by ocfox — APPROVED** [review]

File: pkgs/applications/terminal-emulators/rio/default.nix (hunk 5 above)
Comment (AndersonTorres) [reply pkgs/applications/terminal-emulators/rio/default.nix:34]: Shouldn't it be `Vision`?

https://developer.apple.com/documentation/vision/

**Comment by oluceps** [comment]
How about leaving the broken darwin build and fixing it at another pr, marking it as `broken = stdenv.isDarwin;`.

File: pkgs/applications/terminal-emulators/rio/default.nix (hunk 5 above)
Comment (thiagokokada) [reply pkgs/applications/terminal-emulators/rio/default.nix:34]: Yes, let me try to build this myself locally.

**Review by AndersonTorres — APPROVED** [review]

**Review by otavio — APPROVED** [review]

**Pull request merged**